servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -241,6 +240,8 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
     InodeDirectory currentInodeDirectory = (InodeDirectory) traversalResult.getInode();
     List<Inode> createdInodes = Lists.newArrayList();
     List<Inode> modifiedInodes = Lists.newArrayList();
+    // Directory persistence will not happen until the end of this method.
+    List<Inode> toPersistInodes = Lists.newArrayList(traversalResult.getNonPersisted());
```

Comment:
Are these only directories? Then maybe it should be renamed to <code>toPersistDirectories</code>?